### PR TITLE
feat: Add support for client-side prerequisite events.

### PR DESCRIPTION
**Pull request by kinyoklion**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -320,6 +321,12 @@ function initialize(env, context, specifiedOptions, platform, extraOptionDefs) {
     }
 
     if (sendEvent) {
+      // An event will be send for each of these by virtue of sending events for all flags.
+      if (!allFlags) {
```
Comment (kinyoklion) [inline src/index.js:325]: So, I am not 100% sure there is a good answer here. For sure if you are sending events for all flags, then the prereq will have an event if it was visible to the client.

File: typings.d.ts [hunk 2]
```diff
@@ -552,7 +552,7 @@ declare module 'launchdarkly-js-sdk-common' {
 
   /**
    * Describes the reason that a flag evaluation produced a particular value. This is
-   * part of the {@link LDEvaluationDetail} object returned by {@link LDClient.variationDetail]].
```
Comment (kinyoklion) [inline typings.d.ts:555]: Noticed this when working on the docs for node-client-sdk. So a good chance to just fix that link.

File: src/__tests__/LDClient-events-test.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// @ts-nocheck
```
Comment (kinyoklion) [inline src/__tests__/LDClient-events-test.js:1]: Added this ts-nocheck as typescript checking in eslint isn't happy in this file. Nothing changed, but the typescript version was updated. Basically just affects the IDE.

File: src/index.js (hunk 1 above)
Comment (kinyoklion) [reply src/index.js]: This comment is still relevant github. I just corrected part of the line. >.<

File: src/index.js [hunk 4]
```diff
@@ -50,14 +50,14 @@ function initialize(env, context, specifiedOptions, platform, extraOptionDefs) {
...
     ? diagnostics.DiagnosticsManager(
-        platform,
-        persistentStorage,
-        diagnosticsAccumulator,
-        eventSender,
-        environment,
-        options,
-        diagnosticId
-      )
+      platform,
```
Comment (kinyoklion) [inline src/index.js:53]: There are a few formatting changes from prettier.

**Review by tanderson-ld — APPROVED** [review]

**Pull request merged**